test/framework/src/main/java/org/elasticsearch/script/MockScriptPlugin.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.ScriptPlugin;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * A script plugin that uses {@link MockScriptEngine} as the script engine for tests.
+ */
+public class MockScriptPlugin extends Plugin implements ScriptPlugin {
+
+    public static final String NAME = MockScriptEngine.NAME;
+
+    @Override
+    public ScriptEngineService getScriptEngineService(Settings settings) {
+        return new MockScriptEngine(pluginScriptLang(), pluginScripts());
+    }
+
+    protected Map<String, Function<Map<String, Object>, Object>> pluginScripts() {
+        return Collections.emptyMap();
```

Comment:
In fact I expect to move many more tests to `MockScriptPlugin` and I thought these tests would mostly use the empty definition... But all the tests I moved until now require custom scripts so yes I agree it can be abstract.